app/src/org/commcare/utils/DeviceIdentifier.java

```diff
@@ -0,0 +1,41 @@
+package org.commcare.utils;
+
+import android.content.Context;
+import android.content.SharedPreferences;
+import androidx.preference.PreferenceManager;
+import java.util.UUID;
+
+/**
+ * A Utility to get a unique identifier for app's installation.
+ * @author $|-|!˅@M
+ */
+public final class DeviceIdentifier {
```

Comment:
@ctsims  Goint to merge this to go ahead with code freeze though can you review this class and see the device identifier makes sense to you.